CONTRIBUTING.md

```diff
@@ -10,15 +10,24 @@ Please follow the [code of conduct](CODE_OF_CONDUCT.md) in all your interactions
 
 1. Read [getting started docs](docs/getting_started.md) and prepare Thanos.
 2. Familarize yourself with [Makefile](Makefile) commands like `format`, `build`, `proto` and `test`.
-3. Keep PRs as small as possible. Chain them if needed (base PR on other PRs).
-4. If you don't have a live object store ready add these envvars to skip tests for these:
+3. Fork improbable-eng/thanos.git and start development from your own folk. Here are sample steps to setup your development enviroment.
+```
+$ mkdir -p $GOPATH/src/github.com/improbable-eng
+$ cd $GOPATH/src/github.com/improbable-eng
+$ git clone https://github.com/<your_github_id>/thanos.git
```

Comment:
or something that @SuperQ mentioned: https://gist.github.com/Chaser324/ce0505fbed06b947d962

I think it is worth to keep something like I suggested (or borrow similar example from the gist) 
```
$ git clone https://github.com/<your_github_id>/thanos.git
$ git add remote upstream https://github.com/improbable-eng/thanos.git
```
...and link this: https://gist.github.com/Chaser324/ce0505fbed06b947d962 for details.

Reply:
That's an interesting option, with this you could replace the mkdir/cd/git with `go get` as well.

I think you would also need to mention `git fetch myfork` to update locally.

You can make the `git push` easier by setting the upstream on the branch:

```console
$ git checkout -b mybranch
$ git push --set-upstream myfork mybranch
```

Reply:
yea, that's my workflow. But what Chaser324 suggests might be easier... up to you (origin as fork and another remote for upstream). It is super easy to accidentally try to push to origin with my option (:

Reply:
Git has supported this workflow for a while now, by setting pushurl on the remote. E.g.

```
$ git clone https://github.com/improbable-eng/thanos.git
$ cd thanos/
$ git remote set-url --push origin git@github.com:<your_github_id>/thanos.git
```

This configures the following workflow:
- pulls come from upstream by default;
- pushes go to your fork;
- (accepted PR would bring the changes to upstream, thereby closing the circle)

You might still want to have a remote for your fork so that you can pull from it (when you're collaborating with someone else on a change).
The upside is, you never accidentally push to the upstream remote which you don't have write access to. You'd still want to develop contributions on a separate branch so to not have conflicts.

Reply:
Thanks for all the comments.
https://gist.github.com/Chaser324/ce0505fbed06b947d962 is exactly the flow we are using in my company. :) I will link it into the doc. 
I think for people try to keep `origin` be the working remote to push is to avoid people pushing into original remote(improbable-eng/thanos.git in this case),  directly.  But both (either origin as working branch or origin as improbable-eng/thanos.git). Personally, I would suggest keeping `origin` as working branch.

Reply:
That's fine. But having upstream remote around is nice as well